# 24caratcarrotdata
Analyses used in publication: ""

1) Folder "Analysis_1_HPLCDat_CA19" contains analyses and output related to QC-filter HPLC data from the CA_2019_732PI trial. 
  Data from these analyses are included in Supplemental Table 3 and Supplemental Figure 3. 
2) Folder "Analysis_2_FigsHPLCdat_CA19" contains analyses and output-related HPLC data was further filtered checking for appropriate HPLC data for each color category.
   Test of normality were completed and Outliers were removed here. 
   The counts of each color score were calculated for supplemental 4.
3) Folder "Analysis_3_HPLCDat_WI18" contains analyses and output related to QC-filter HPLC data from the WI_2018_432 trial.
     Data from these analyses are included in Supplemental Table 2 and Supplemental Figure 1. 
4) Folder "Analysis_4_FigsHPLCdat_WI18" contains analyses and output related contains analyses and output-related HPLC data was further filtered checking for appropriate HPLC data for each color category.
5) Folder "Analysis_5_DescribePhenotypeData_WI18" Removes outliers from the WI_2018_432 trial, and calculates the mean for each carotenoid. Mean separation for Supplemental Figure 2a,b is included here as well as the boxplot in the figure. These analyses include the correlation test found in Table 1. A phenotype file for  Genomic Prediction and GWA analyses was created during these analyses. 
6)  Folder: "Analysis_6_DescribePhenotypeData_CA19" contains analyses and output related to mean separation between colors, and the boxplot found in Supplemental Figure 4 and Supplemental Table 6. The correlation and carotenoid means for Table 1 were calculated with this analysis. Core color carotenoid comparisons found in Supplemental Figure 2 were calculated in this analysis. A comparison of HPLC data (i.e., mean separation and correlation) found in supplemental table 7 was calculated here). 
7) Folder: "Analysis_7_CA19.VCF" contains a text file describing how TASSEL was used to identify variants and filter SNP data.
8) Folder: "Analysis_8_FilterReseq" contains steps to filter the genotyping data (from whole-genome resequencing) for PIs included in the WI_2018_432 trial. This folder also contains the steps used to identify SNPs shared with the CA_2019_732 trial, or a batch of very similar SNPs.
9) Folder: "Analysis_9_ID_SNPs" contains the sets to identify SNPs find equivalent or matching SNPs to those in the WI_2018_432 trial.
10) Folder: "Analysis_10_SubsetSNPs"
11) Folder "Analysis_11_Format2Plink" contains script to convert vcf to plink format. The plink files were used for fastStructure, Admixture, LD decay, and distance matrix calculations.
12) Folder "Analysis_12_fastStructure" contains the batch scripts to complete fastStructure population structure analyses testing 10 populations for both trials.
13) Folder "Analysis_13_Admixture_Plink"
14) Folder "Analysis_14_LDCalculations" contains the steps to determine the rate of LD decay from the plink LD analyses for both the WI_2018_432 and CA_2019_732 trials.
15) Folder "Analysis_15_polyRad"
16) Folder "Analysis_16_fst_Dist" contains analyses used to calculate Fst between populations and summarize the results of the distance matrix as it relates to populations.
17) 
